File: JavaScript-Fundamentals/06.Functions/04.Appearance count/Appearance count.js
/**  Problem 4. Appearance count
 * Write a method that counts how many times given number appears in a given array.
 */
function solve(input) {
    var nums = input[0],
        numbers = input[1].split(' ').map(Number),
        x = +input[2],
        position = 0,
        appCount = 0;

    position = numbers.indexOf(x);

    for (var i = 0; i < nums; i += 1) {

        if (position >= 0) {
            appCount += 1;
        }
        else {
            return appCount;
        }
        position = numbers.indexOf(x, position + 1);
    }
}
// Test
console.log(solve(['8', '28 6 21 6 -7856 73 73 -56', '73']));
console.log(solve(['8', '73 6 21 6 -7856 73 73 -56', '73']));

File: Homeworks/1. Programming/2. C#-Part-2/03.Methods/04.Appearance count/AppearanceCount.cs
//Write a method that counts how many times given number appears in a given array.
namespace _04.Appearance_count
{
    using System;
    using System.Linq;

    class AppearanceCount
    {
        static int CountApperance(int[] numbers, int toFind)
        {
            int counter = 0;

            foreach (var item in numbers)
            {
                if(item==toFind)
                {
                    counter++;
                }
            }

            return counter;
        }

        static void Main()
        {
            int input = int.Parse(Console.ReadLine());
            var numbers = Console.ReadLine().Split(' ').Select(x => int.Parse(x)).ToArray();
            int toFind = int.Parse(Console.ReadLine());
            int count = CountApperance(numbers,toFind);

            Console.WriteLine(count);
        }
    }
}
